dot/rpc/modules/system.go

```diff
@@ -73,8 +75,9 @@ func (sm *SystemModule) Name(r *http.Request, req *EmptyRequest, res *string) er
 
 // ChainType returns the chain type
 func (sm *SystemModule) ChainType(r *http.Request, req *EmptyRequest, res *string) error {
-	*res = sm.systemAPI.ChainType()
-	return nil
+	gd, err := sm.storageAPI.GetGenesisData()
+	*res = gd.ChainType
```

Comment:
We should assign if err is `nil`

Reply:
I moved initialization to dot createSystemService, and now I'm catching the error there.